zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -1572,12 +1579,13 @@ object JsonCodec {
             buffer(idx) = field.defaultValue.get
           } else {
             val schema = field.schema match {
-              case l @ Schema.Lazy(_) => l.schema
+              case l: Schema.Lazy[_] => l.schema
```

Comment:
Does this make a difference?

Reply:
Yes, It does. It allows to avoid `.unapply` call in Scala 3 that allocates a redundant instance of option of Tuple1.

Reply:
In the last commit beside formatting I fixed redundant 2 variables generated by Scala 3 compiler for that place. Now the byte code decompiled to Java looks more clear for that piece of code:
```java
                        Schema schema = field.schema();
                        if (schema instanceof Schema.Lazy) {
                            Schema.Lazy l = (Schema.Lazy)schema;
                            schema = l.schema();
                        }
```